### ISSUE TITLE
Dataset and IOU calculation issues

### ISSUE THREAD

**Issue by MmelodYy** [UCSB-AI/Aerial-Vision-and-Dialog-Navigation#22]
Hello, author. When I checked the dataset and evaluation metrics, I found several issues. I would like to confirm with you whether my checks are incorrect, as this will affect the entire experiment.

The specific issues are as follows:
(1) I repeatedly compared the text descriptions with the actual trajectories calculated by GPS, and I found that some data descriptions seem to be inconsistent (whether they are centered on the self or on absolute orientation). [I also know that the GPS coordinate system and the image coordinate system are different, so I believe there is no issue with my checking process.]

(2) In the test seen dataset, I found that there are some scenes that were not seen in the training set.

(3) Due to the ambiguity of height information, IOU calculation does not seem to serve as a termination condition. When the heights are different, the visible area of the drone varies, which may lead to containment relationships, rendering the IOU calculation ineffective. (Although the ideal IOU should be calculated based on similar heights.)

(4) There seems to be an issue with the calculation of the GP metric. Shouldn't we use path[-1] instead of path[-2]?

(5) Because this is a navigation system based on aerial views, it simulates a 3D scene. Height information seems impossible to derive from the known path, so in an unknown environment, this will inevitably affect the model's perception of the actual next height. (Although the train_data has teacher forcing guidance, this does not mean that the model can perceive heights as in the test unseen data, which will undoubtedly affect its generalization performance.)

I also found many potential issues, but I won't elaborate on each one here. However, based on the problems mentioned above, I believe this will affect the research of many others in the future, so I want to express my concerns. Thank you very much.

**Comment by UeFan** [comment]
Hi, thanks for bringing out these questions.

1. The descriptions are human written, although we had made efforts to check them, they are not 100% error free. However, despite the inaccurate desciptions, the drone reaches the destination at the end of the corresponding dialog. This shows the  key information is in those text.
2. I think this is not very likely. Could you further identify them? Seen dataset means the data of the drone observing scenes sampled from the xView satellite images that are also used in training data. They may not be exatly the same scenes, but they are geographically close and sharing the samilar look.
3. I think you have some misunderstanding about the IOU termination condition. As the drone flying higher, the view of the drone will be larger. So for an example, when the destination is a small residential house, and the drone is hovering at 400m hight in the air, right above the house. At this time step, the drone's view will be centered at the ground truth destination area, but much larger than the destination area. In this case, the IOU beween them is small and the drone navigation is not succeed at this case.
4. For this problem, please refer to https://github.com/eric-ai-lab/Aerial-Vision-and-Dialog-Navigation/issues/12
5. In AVDN, the drone's view is simulated by satellite images. During the drone's navigation, the drone will observe a small segment of the satellite image. We simulate the drone's view change when the drone's height change by zoom-in or zoom-out the satellite images, which means that the drone's view size will be changed on the satellite images.

Hope my answer will help clear your questions and please do not hesitate to ask more.

Thanks.

**Comment by MmelodYy** [comment]
Firstly, thank you very much for your reply. At the same time, I fully understand your contribution to that work. Because of the amount of time I have invested in that task, that is why I have found so many questions that have puzzled me. Next, back to the specific question level.

1. regarding the accuracy of the text describing the directional words, I did compare them from a number of perspectives, and it does seem that there are quite a few labelling errors. Introducing human attention aera annotation does effectively alleviate the problem, but I think the incorrect text annotation is more of a limitation on its upper performance limit. (After all, the model mainly learns the laws of such textual descriptions, even if they do not correspond to the laws of real orientation descriptions.)

2. as for the problem with the val seen dataset, the point I was trying to make is that I found that some of the maps used in that dataset are not present inside the train. That is to say, some of the maps used for the paths should be counted as part of the unseen dataset. (So I guess this is one of the reasons why the performance on the test seen dataset is lower than on the unseen dataset.)
 
3. my main concern is still the highly simulated problem. Since I've invested several months in this problem, I think I should have a relatively deep understanding of what a lot of the code is saying. I know author that you are going to adjust the height by zoom operation. And my point of concern is that there is no implicit height adjustment information given in the text description, so theoretically the model cannot learn the final correct height. (As an analogy to the AerialVLN dataset, I found that their text descriptions are suggestive of ‘stopping at a building’, and they are using a 3D simulator.) Changes in field of view do represent changes in height, but this does not imply that the final height should be stopped.In short, I found out about these problems during the experiment, and even when I was writing my thesis to collate the data. So what comes next will affect my decision on whether to continue writing or to go ahead and solve the above problems. (Because I feel that even with 2D remote sensing images, the above problems, especially the height problem, can still be solved equally effectively. I have also made some attempts and have now initially verified the feasibility.) 

Finally, I would like to express that I am very grateful for the author's work open source, and I am very sure of your contribution to the development of this field.

Based on this data set will influence a lot of people to take into consideration, as well as for the author. So I closed the issue at that time. thank you very much for the author's understanding.

**Comment by UeFan** [comment]
Thanks for the questions. I am glad you point out these problems of the work and I really admire your comments. This work was the first one exploring the drone VLN via dialog, so please forgive us for the many limitations. 

For problem 1 and 3, I think you are mainly puzzled by the data quality and if a model can rely on AVDN training data to conquer the task. I think the answer is negative as the issue you have pointed out: 1) human annotation mistakes and 3) ambiguous supervision signal for the drone height control. So I think these are all valuable directions to investigate in order to build better Drone VLN models. I would suggest to filter the training data before using it and I believe that more efforts are required to find out how to train the drone to learn the height adjusting. 
For problem 2, I think you have the point and I think this is truly a bug in the dataset building process. If you are interested, I would appreciate it very much if you can contribute to AVDN by builidng a new seen val set by removing those unseen data.

Thank you.

**Comment by MmelodYy** [comment]
Thank you very much for your reply. You have encouraged me to continue on this path. I will work on some of the potential problems with AVDN in my follow up work.

Finally, thank you again for your response and your contribution to the field.
